src/world.cpp

```diff
@@ -69,6 +69,50 @@ void world_spawn_npcs(void *world_void)
 	}
 }
 
+void world_npc_speech(NPC* from)
+{
+	const auto& data = from->Data();
+
+	if (data.talk_phrases.empty())
+	{
+		return;
+	}
+
+	if (from->last_talk + data.talk_speed > Timer::GetTime())
+	{
+		return;
+	}
+
+	from->last_talk = Timer::GetTime();
```

Comment:
Yes. As a server owner, I want to be able to say "Every {frequency} seconds, the NPC has a {percentage} chance of saying {phrases}".
That's just my take on it. Thoughts?

Reply:
After rereading the implementation I think the issue I was thinking of isn't possible unless you define speech for an NPC that isn't on the map, which is fine because then it _should_ never say anything.